src/uct/ib/ud/base/ud_ep.c

```diff
@@ -1149,6 +1156,10 @@ ucs_status_t uct_ud_ep_pending_add(uct_ep_h ep_h, uct_pending_req_t *req,
         goto add_req;
     }
 
+    if (ucs_unlikely(flags & UCT_CB_FLAG_ASYNC)) {
```

Comment:
why we don't check resources when adding pending async? there could be reorder when it's called from main thread

Reply:
Discussed in https://github.com/openucx/ucx/pull/2864#discussion_r220528239
> It's needed to make possible adding to pending from async am response callback. (uct_ud_iface_has_skbs() returns 0)

Reply:
so can we also check if async flag is raised?
the condition here must be equivalent to do_pending